testdata/schema.go

```diff
@@ -0,0 +1,232 @@
+	primitiveTypesType := reflect.TypeOf(arrow.PrimitiveTypes)
+	fields := make([]arrow.Field, primitiveTypesType.NumField())
+	for i := 0; i < primitiveTypesType.NumField(); i++ {
+		fieldName := primitiveTypesType.Field(i).Name
+		dataType := primitiveTypesValue.FieldByName(fieldName).Interface().(arrow.DataType)
+		fields[i] = arrow.Field{Name: strings.ToLower(fieldName), Type: dataType, Nullable: true}
+	}
+	return fields
+}
+
+// BinaryFields returns a list of binary fields as defined by Arrow.
+func BinaryFields() []arrow.Field {
+	binaryTypesValue := reflect.ValueOf(arrow.BinaryTypes)
+	binaryTypesType := reflect.TypeOf(arrow.BinaryTypes)
+	fields := make([]arrow.Field, binaryTypesType.NumField())
+	for i := 0; i < binaryTypesType.NumField(); i++ {
+		fieldName := binaryTypesType.Field(i).Name
+		dataType := binaryTypesValue.FieldByName(fieldName).Interface().(arrow.DataType)
+		fields[i] = arrow.Field{Name: strings.ToLower(fieldName), Type: dataType, Nullable: true}
+	}
+	return fields
+}
+
+// FixedWidthFields returns a list of fixed width fields as defined by Arrow.
+func FixedWidthFields() []arrow.Field {
+	fixedWidthTypesValue := reflect.ValueOf(arrow.FixedWidthTypes)
+	fixedWidthTypesType := reflect.TypeOf(arrow.FixedWidthTypes)
+	fields := make([]arrow.Field, fixedWidthTypesType.NumField())
+	for i := 0; i < fixedWidthTypesType.NumField(); i++ {
+		fieldName := fixedWidthTypesType.Field(i).Name
+		dataType := fixedWidthTypesValue.FieldByName(fieldName).Interface().(arrow.DataType)
+		fields[i] = arrow.Field{Name: strings.ToLower(fieldName), Type: dataType, Nullable: true}
+	}
+	return fields
+}
+
+func sortAndRemoveDuplicates(fields []arrow.Field) []arrow.Field {
+	newFields := make([]arrow.Field, len(fields))
+	copy(newFields, fields)
+	sort.Slice(newFields, func(i, j int) bool {
+		return newFields[i].Name < newFields[j].Name
+	})
+	for i := 0; i < len(newFields)-1; i++ {
+		if newFields[i].Name == newFields[i+1].Name {
+			newFields = append(newFields[:i], newFields[i+1:]...)
+			i--
+		}
+	}
+	return newFields
+}
+
+func removeFieldsByType(fields []arrow.Field, t ...arrow.Type) []arrow.Field {
+	newFields := make([]arrow.Field, len(fields))
+	copy(newFields, fields)
+	for _, d := range t {
+		for i := 0; i < len(newFields); i++ {
+			if newFields[i].Type.ID() == d {
+				newFields = append(newFields[:i], newFields[i+1:]...)
+				i--
+			}
+		}
+	}
+	return newFields
+}
+
+func removeFieldsByDataType(fields []arrow.Field, dt ...arrow.DataType) []arrow.Field {
+	newFields := make([]arrow.Field, len(fields))
+	copy(newFields, fields)
+	for _, d := range dt {
+		for i := 0; i < len(newFields); i++ {
+			if arrow.TypeEqual(newFields[i].Type, d) {
+				newFields = append(newFields[:i], newFields[i+1:]...)
+				i--
+			}
+		}
+	}
+	return newFields
+}
+
+// ListOfFields returns a list of fields that are lists of the given fields.
+func ListOfFields(baseFields []arrow.Field) []arrow.Field {
```

Comment:
Large lists, fixed size lists?

Reply:
Good point, but I think we can add these when the need arises? Right now adding a new type requires handling in every destination, so we shouldn't add a new type to the tests unless there is a need from it from a source plugin. I'm not aware of sources that would need large lists or fixed size lists.